test/benchmark/sidb_logic.cpp

```diff
@@ -0,0 +1,268 @@
+//
+// Created by Willem Lambooy on 19/02/2025.
+//
+
+#include <catch2/benchmark/catch_benchmark.hpp>
+#include <catch2/catch_test_macros.hpp>
+
+#include "../utils/blueprints/layout_blueprints.hpp"
+
+#include <fiction/algorithms/physical_design/design_sidb_gates.hpp>
+#include <fiction/algorithms/simulation/sidb/is_operational.hpp>
+#include <fiction/types.hpp>
+
+using namespace fiction;
+
+TEST_CASE("Benchmark exact operational assessment", "[benchmark]")
```

Comment:
@wlambooy would it be possible to also test the different options for ``strategy_to_analyze_operational_status``? Many thanks! 🙏